source/Irrlicht/CBAWMeshFileLoader.cpp

```diff
@@ -0,0 +1,274 @@
+// Copyright (C) 2018 Krzysztof "Criss" Szenk
+// This file is part of the "Irrlicht Engine" and "Build A World".
+// For conditions of distribution and use, see copyright notice in irrlicht.h
+// and on http://irrlicht.sourceforge.net/forum/viewtopic.php?f=2&t=49672
+
+#include "CBAWMeshFileLoader.h"
+
+#include "CFinalBoneHierarchy.h"
+#include "SMesh.h"
+#include "CSkinnedMesh.h"
+
+namespace irr { namespace scene
+{
+CBAWMeshFileLoader::~CBAWMeshFileLoader()
+{
+	if (m_fileSystem)
+		m_fileSystem->drop();
+}
+
+CBAWMeshFileLoader::CBAWMeshFileLoader(scene::ISceneManager* _sm, io::IFileSystem* _fs) : m_sceneMgr(_sm), m_fileSystem(_fs)
+{
+#ifdef _DEBUG
+	setDebugName("CBAWMeshFileLoader");
+#endif
+	if (m_fileSystem)
+		m_fileSystem->grab();
+}
+
+template<>
+core::ICPUBuffer* CBAWMeshFileLoader::make<core::ICPUBuffer>(SBlobData& _data, SContext& _ctx) const
+{
+	if (_data.wasCreated)
+		return dynamic_cast<core::ICPUBuffer*>(_ctx.createdObjects[_data.header->handle]);
+
+	typename core::CorrespondingBlobTypeFor<core::ICPUBuffer>::type* blob = core::toBlobPtr<core::ICPUBuffer>(_data.blob);
+
+	void* blobData = malloc(_data.header->blobSizeDecompr);
+	memcpy(blobData, blob->getData(), _data.header->blobSizeDecompr);
+	core::ICPUBuffer* buf = new core::ICPUBuffer(_data.header->blobSizeDecompr, blobData);
```

Comment:
create a new core::ICPUBuffer(size) without a pointer and it will malloc the storage for you.

then do ctx.file->read() directly into the buf->getData()